core/src/main/java/feast/core/training/BigQueryDatasetTemplater.java

```diff
@@ -117,9 +119,11 @@ private String formatDateString(Timestamp timestamp) {
     final List<String> columns;
     final String tableId;
 
-    public Features(List<String> featureIds, String tableId) {
-      this.columns = featureIds.stream()
-          .map(f -> f.replace(".", "_"))
+    Features(List<String> featureIds, String tableId, String entityName) {
+      // columns represent the column name in BigQuery
+      // feature with id "myentity.myfeature" will be represented as column "myfeature" in BigQuery
+      columns = featureIds.stream()
+          .map(f -> f.replace(".", "_").replace(entityName + "_", ""))
```

Comment:
I think rather than do this odd replace of the event name + "_", it would be neater if you we were to derive featureNames, from the featureInfos.

eg:
```
List<String> featureNames = featureInfos.stream().map(FeatureInfo::getName).collect(Collectors.toSet());
```

Reply:
Agree, but be careful that feature name might also contain "." which might need to be replaced with "_" or other BQ friendly character.

Reply:
Yeah sounds good. I made the update to use FeatureInfo to assign columns value. 

For checking whether feature name may contain invalid characters, Aria and I think in practice, this is unlikely to happen, because the feature spec registration will have failed in the first place.